### ISSUE TITLE
RF01 & AM08: Unexpected failure for `join unnest with offset` in BigQuery

### ISSUE THREAD

**Issue by lishenjie** [sqlfluff/sqlfluff#6725]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

The following BQ query fails RF01 & AM08.

It's possibly after #6400 and #6239.

### Expected Behaviour

```
All Finished 📜 🎉!
```

### Observed Behaviour

```
== [test.sql] FAIL                                                                                                                                                                                                                                           
L:   2 | P:   5 | RF01 | Reference 'ix' refers to table/view not found in the
                       | FROM clause or found in ancestor statement.
                       | [references.from]
L:   5 | P:   1 | AM08 | Implicit cross join detected.
                       | [ambiguous.join_condition]
All Finished 📜 🎉!
```

`ix` is the offset alias (https://cloud.google.com/bigquery/docs/arrays).
`left join` is to include rows with `null` `value_list` arrays (https://cloud.google.com/bigquery/docs/arrays).

### How to reproduce

```sqlfluff lint test.sql --dialect bigquery```

test.sql:
```sql
select
    ix,
    v
from t
left join
    unnest(t.value_list) as v
    with offset as ix
```

### Dialect

bigquery

### Version

sqlfluff, version 3.3.0
Python 3.11.11

### Configuration

The default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
